### PR TITLE
Role/Permission Endpoints

### PR DESCRIPTION
**Pull request by lboeman**
Adds User, Role and Permissions endpoints to the API. User creation/deletion endpoints are just method stubs and documentation at the moment to get feedback on the layout of the endpoints, I noted the procedures that each endpoint would end up calling.
I added the User->Role mapping functions to Users, but tagged them with both 'Users' and 'Roles'  so that they can be found in both sections of the documentation, this could be repeated for the Roles -> Permissions mappings.

A couple of things I noted while working on this:
- There is no `delete_user` procedure, should that exist or is that going to be something we defer to auth0?
- Should auth0_id be treated as a secret, and the internal SolarForecastArbiter UUID used to identify users? other way around? neither?

### CODE REVIEW

File: sfa_api/schema.py [hunk 1]
```diff
@@ -526,3 +526,39 @@ class CDFForecastGroupSchema(CDFForecastGroupPostSchema):
...
+@spec.define_schema('UserPostSchema')
+class UserPostSchema(ma.Schema):
+    auth0_id = ma.String(
+        title="Auth0 ID",
+        description="The User's unique Auth0 identifier.",
+    )
+    # Not sure if this should be here, on one hand we want Admin at an
+    # Organization to only be able to add Users for their organization,
+    # but framework administrators would need a way to add a user to any
+    # organization.
```
Comment (alorenzo175) [inline sfa_api/schema.py]: For now, I would leave off user creation/deletion. They will require integration with Auth0 and I'm not sure how that will work yet.

File: sfa_api/schema.py [hunk 2]
```diff
@@ -526,3 +526,39 @@ class CDFForecastGroupSchema(CDFForecastGroupPostSchema):
...
+@spec.define_schema('UserSchema')
+class UserSchema(ma.Schema):
+    _links = ma.Hyperlinks(
+        {
+            'roles': ma.AbsoluteURLFor('roles.user_roles',
+                                       user_id='<user_id>'),
+        },
+        description="Contains a link to the Users associated Roles."
+    )
+    # Should auth0_id be included in the response schema?
```
Comment (alorenzo175) [inline sfa_api/schema.py]: probably just uuid

File: sfa_api/schema.py [hunk 3]
```diff
@@ -526,3 +526,39 @@ class CDFForecastGroupSchema(CDFForecastGroupPostSchema):
...
+    user_id = ma.UUID(
+        title="User ID",
+        description="Unique UUID of the User.",
+    )
+    organization_id = ma.UUID(
+        title="Organization ID",
+        description="UUID of the Organization the User belongs to."
+    )
```
Comment (alorenzo175) [inline sfa_api/schema.py]: not sure if this will have more information retrieved from auth0

**Comment by lboeman** [comment]
@alorenzo175  While working on this, I realized the users/roles/permissions procedures are defined with other MYSQL users. Is the MYSQL user 'apiuser' meant to have grants to execute the CRUD procedures for those types of objects? I added those permissions locally for testing by hand while working on this, but I wasn't sure if there is another plan for that?

**Comment by alorenzo175** [comment]
Yes, apiuser should have the execute privileges on the functions the python code uses

**Comment by lboeman** [comment]
Opened a couple of further issues I found while working on this #122 and #123 . They are not breaking issues, but are unexpected behavior.

File: sfa_api/conftest.py [hunk 4]
```diff
@@ -144,12 +144,19 @@ def _make_sql_app():
             yield app
 
 
-@pytest.fixture(scope='module')
-def sql_app():
+@pytest.fixture()
+def sql_app(mocker):
```
Comment (alorenzo175) [inline sfa_api/conftest.py]: why change this? maybe leftover from when you where trying to get permissions to work properly

File: sfa_api/schema.py [hunk 5]
```diff
@@ -526,3 +529,98 @@ class CDFForecastGroupSchema(CDFForecastGroupPostSchema):
...
+        validate=validate.OneOf(['create', 'read', 'update',
+                                 'delete', 'read_values', 'write_values',
+                                 'delete_values']),
+    )
+    object_type = ma.String(
+        title="Object Type",
+        description="The type of object this permission will act on.",
+        required=True,
+        validate=validate.OneOf(['sites', 'aggregates', 'forecasts',
+                                 'observations', 'users', 'roles',
```
Comment (alorenzo175) [inline sfa_api/schema.py]: need cdf_forecasts

**Pull request merged**